noether/src/segmentation_server.cpp

```diff
@@ -123,6 +146,9 @@ class SegmentationAction
     ros::Time tStart = ros::Time::now();
     segmenter.segmentMesh();
```

Comment:
You may want to run the segmentation function from a different thread or from a ros::Timer in order to prevent the node from blocking the calling client.

Reply:
Two things. 
1) Apparently SimpleActionServer handles accepting goals automatically (see [here](https://docs.ros.org/kinetic/api/actionlib/html/classactionlib_1_1SimpleActionServer.html)).
2) The callback always blocks the thread it is running on. So in order to avoid that, use the callback client, and make sure you are using an async spinner. I made a note about that. Interestingly, it appears that sendgoal is blocking if you define only the done callback and not the active and feedback. I'm not sure why.

Neither of these things really effect this example that much. But they are useful for me to know when writing our clients.